src/test/lrsql/ops/query/reaction_test.clj

```diff
@@ -0,0 +1,217 @@
+          trigger-id (u/str->uuid (get tc/reaction-stmt-b "id"))
+          {reaction-id
+           :result}  (adp/-create-reaction
+                      lrs "reaction-0" tc/simple-reaction-ruleset true)]
+      ;; Add statements
+      (doseq [s [tc/reaction-stmt-a
+                 tc/reaction-stmt-b]]
+        (Thread/sleep 100)
+        (lrsp/-store-statements lrs tc/auth-ident [s] []))
+      (try
+        (is (= {:result
+                [{:reaction-id reaction-id
+                  :trigger-id  trigger-id
+                  :statement
+                  {"actor"   {"mbox" "mailto:bob@example.com"},
+                   "verb"    {"id" "https://example.com/verbs/completed"},
+                   "object"
+                   {"id"         "https://example.com/activities/a-and-b",
+                    "objectType" "Activity"}
+                   "context" {"extensions" {"https://example.com/foo" nil}}}
+                  ;; Authority derived from the trigger statement
+                  :authority   (:agent tc/auth-ident)}]}
+               (qr/query-statement-reactions
+                bk ds {:trigger-id trigger-id})))
+        (finally (component/stop sys'))))))
+
+(deftest query-statement-reactions-custom-authority-test
+  (testing "Valid reaction with custom authority"
+    (let [sys        (support/test-system
+                      :conf-overrides
+                      {[:lrs :enable-reactions] false})
+          sys'       (component/start sys)
+          lrs        (-> sys' :lrs)
+          bk         (:backend lrs)
+          ds         (-> sys' :lrs :connection :conn-pool)
+          trigger-id (u/str->uuid (get tc/reaction-stmt-b "id"))
+          {reaction-id
+           :result}  (adp/-create-reaction
+                      lrs
+                      "reaction-0"
+                      (assoc-in
+                       tc/simple-reaction-ruleset
+                       [:template "authority"]
+                       {"$templatePath" ["a" "actor"]})
+                      true)]
+      ;; Add statements
+      (doseq [s [tc/reaction-stmt-a
+                 tc/reaction-stmt-b]]
+        (Thread/sleep 100)
+        (lrsp/-store-statements lrs tc/auth-ident [s] []))
+      (try
+        (is (= {:result
+                [{:reaction-id reaction-id
+                  :trigger-id  trigger-id
+                  :statement
+                  {"actor"     {"mbox" "mailto:bob@example.com"},
+                   "verb"      {"id" "https://example.com/verbs/completed"},
+                   "object"
+                   {"id"         "https://example.com/activities/a-and-b",
+                    "objectType" "Activity"}
+                   "context"   {"extensions" {"https://example.com/foo" nil}}
+                   "authority" {"mbox" "mailto:bob@example.com"}}
+                  ;; Authority derived from the template
+                  :authority   {"mbox" "mailto:bob@example.com"}}]}
```

Comment:
And here

Reply:
a3b9b77d9ca7e227154d5199f40d251015332653